sdk/cosmos/src/operations/delete_document.rs

```diff
@@ -0,0 +1,85 @@
+use crate::prelude::*;
+use crate::responses::DeleteDocumentResponse;
+use crate::headers::from_headers::*;
+use azure_core::prelude::*;
+use azure_core::headers::session_token_from_headers;
+
+use chrono::{DateTime, Utc};
+use http::StatusCode;
+use http::response::Response;
+use std::convert::TryInto;
+
+#[derive(Debug, Clone)]
+pub struct DeleteDocumentOptions<'a> {
+    document_client: &'a DocumentClient,
+    if_match_condition: Option<IfMatchCondition<'a>>,
+    if_modified_since: Option<IfModifiedSince<'a>>,
+    activity_id: Option<ActivityId<'a>>,
+    consistency_level: Option<ConsistencyLevel>,
+    allow_tentative_writes: TenativeWritesAllowance,
```

Comment:
The `document_client` and `activity_id` fields are not needed and can be removed.